tests/tests/test_fault_tolerance.py

```diff
@@ -222,9 +222,30 @@ def test_image_download_retry_timeout(
 
         with mender_device.get_reboot_detector(host_ip) as reboot:
             if test_set["blockAfterStart"]:
-                # Block after we start the download.
-                deployment_id, new_yocto_id = common_update_procedure(valid_image)
-                mender_device.run("fuser -mv %s" % (inactive_part))
+                # Use an state script to pause the deployment so that
+                # the test can manipulate the network half way.
+                script_content = """#!/bin/sh
+
+set -e
+touch /tmp/wait-for-install
+while [ -f /tmp/wait-for-install ]; do
+    echo Waiting...
+    sleep 10
+done
+exit 0"""
+                try:
+                    f = open("ArtifactInstall_Enter_00", "w")
+                    f.write(script_content)
+                    f.close()
+
+                    deployment_id, new_yocto_id = common_update_procedure(
+                        valid_image, scripts=["ArtifactInstall_Enter_00"]
+                    )
+                finally:
+                    os.unlink(f.name)
+
+                # Wait until the update is in progress
+                mender_device.run("test -f /tmp/wait-for-install")
```

Comment:
Are you really waiting though? `run()` will retry failed connections, but not failed commands, or?

Reply:
Yes, it "waits" in the sense that it retries the command until returns 0. See output below:
```
[gw0] TestFaultTolerance__test_image_download_retry_timeout[test_set0] -- 2021-07-08 09:56:20,925 [INFO]: >> Got UnexpectedExit while executing command in host 172.18.0.2: Encountered a bad command exit code!

Command: 'test -f /tmp/wait-for-install'

Exit code: 1

Stdout: already printed

Stderr: already printed


[gw0] TestFaultTolerance__test_image_download_retry_timeout[test_set0] -- 2021-07-08 09:56:22,942 [INFO]: >> Got UnexpectedExit while executing command in host 172.18.0.2: Encountered a bad command exit code!

Command: 'test -f /tmp/wait-for-install'

Exit code: 1

Stdout: already printed

Stderr: already printed


[gw0] TestFaultTolerance__test_image_download_retry_timeout[test_set0] -- 2021-07-08 09:56:27,061 [INFO]: >> Got UnexpectedExit while executing command in host 172.18.0.2: Encountered a bad command exit code!

Command: 'test -f /tmp/wait-for-install'

Exit code: 1

Stdout: already printed

Stderr: already printed


[gw0] TestFaultTolerance__test_image_download_retry_timeout[test_set0] -- 2021-07-08 09:56:36,187 [INFO]: >> Disallowing network communication to docker.mender.io
...
```

Reply:
Oh, I didn't know this, actually. It's a little... unexpected, I would say. Is it clearer to use a while loop in the shell command instead? This also saves you from having to redo the SSH handshake, and produces less scary output.

Whether it solves the problem, you had, well that's another matter. Probably not.